packages/astro/src/content/vite-plugin-content-assets.ts

```diff
@@ -73,42 +76,95 @@ export function astroContentAssetPropagationPlugin({ mode }: { mode: string }):
 }
 
-export function astroContentProdBundlePlugin({ internals }: { internals: BuildInternals }): Plugin {
-	return {
-		name: 'astro:content-prod-bundle',
-		async generateBundle(_options, bundle) {
-			for (const [_, chunk] of Object.entries(bundle)) {
-				if (chunk.type === 'chunk' && chunk.code.includes(LINKS_PLACEHOLDER)) {
-					for (const id of Object.keys(chunk.modules)) {
-						for (const [pageInfo, depth, order] of walkParentInfos(id, this)) {
-							if (moduleIsTopLevelPage(pageInfo)) {
-								const pageViteID = pageInfo.id;
-								const pageData = getPageDataByViteID(internals, pageViteID);
-								if (!pageData) continue;
-								const entryCss = pageData.contentCollectionCss?.get(id);
-								if (!entryCss) continue;
-								chunk.code = chunk.code.replace(
-									JSON.stringify(LINKS_PLACEHOLDER),
-									JSON.stringify([...entryCss])
-								);
-							}
-						}
-					}
-				}
-			}
-		},
-	};
-}
-
-export function astroConfigBuildPlugin(internals: BuildInternals): AstroBuildPlugin {
+export function astroConfigBuildPlugin(options: StaticBuildOptions, internals: BuildInternals): AstroBuildPlugin {
+	let ssrPluginContext: any = undefined;
 	return {
 		build: 'ssr',
 		hooks: {
-			'build:before': () => {
+			'build:before': ({ build }) => {
 				return {
-					vitePlugin: astroContentProdBundlePlugin({ internals }),
+					vitePlugin: {
+						name: 'astro:content-build-plugin',
+						generateBundle() {
+							if(build === 'ssr') {
+								ssrPluginContext = this;
+							}
+						}
+					},
 				};
 			},
+			'build:post': ({ ssrOutputs, clientOutputs, mutate }) => {
+				const outputs = ssrOutputs.flatMap(o => o.output);
+				for (const chunk of outputs) {
+					if (
+						chunk.type === 'chunk' &&
+						(chunk.code.includes(LINKS_PLACEHOLDER) || chunk.code.includes(SCRIPTS_PLACEHOLDER))
+					) {
+						let entryCSS = new Set<string>();
+						let entryScripts = new Set<string>();
+
+						for (const id of Object.keys(chunk.modules)) {
+							for (const [pageInfo] of walkParentInfos(id, ssrPluginContext)) {
+								if (moduleIsTopLevelPage(pageInfo)) {
+									const pageViteID = pageInfo.id;
+									const pageData = getPageDataByViteID(internals, pageViteID);
+									if (!pageData) continue;
+	
+									const _entryCss = pageData.contentCollectionCss?.get(id);
+									const _entryScripts = pageData.propagatedScripts?.get(id);
+									if(_entryCss) {
+										for(const value of _entryCss) {
+											entryCSS.add(value);
+										}
+									}
+									if(_entryScripts) {
+										for(const value of _entryScripts) {
+											entryScripts.add(value);
+										}
+									}
+								}
+							}
+						}
+
+						let newCode = chunk.code;
+						if (entryCSS.size) {
+							newCode = newCode.replace(
+								JSON.stringify(LINKS_PLACEHOLDER),
+								JSON.stringify([...entryCSS])
+							);
+						}
+						if (entryScripts.size) {
+							const entryFileNames = new Set<string>();
```

Comment:
_Love_ that we can look these up now 👏 Way simpler than scary naming conventions!